cpp/include/graph_generators.hpp

```diff
@@ -0,0 +1,314 @@
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param num_vertices Number of vertices to use in each complete graph
+ * @param num_graphs Number of graphs to generate
+ * @param symmetrize If true, symmetrize the edges
+ * @return std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>> A tuple of
+ * rmm::device_uvector objects for edge source vertex IDs and edge destination vertex IDs.
+ */
+template <typename vertex_t>
+std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>>
+generate_complete_graph_edgelist(raft::handle_t const& handle,
+                                 size_t num_vertices,
+                                 size_t num_graphs,
+                                 bool symmetrize = false);
+
+/**
+ * @brief generate an edge lists for an Erdos-Renyi graph
+ *
+ * If executed in a multi-gpu context (handle comms has been initialized)
+ * each GPU will generate Erdos-Renyi edges for its portion of the 2D
+ * partitioning of the adjacency matrix.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param num_vertices Number of vertices to use in each complete graph
+ * @param p Probability for edge creation
+ * @param symmetrize If true, symmetrize the edges
+ * @param seed Seed value for the random number generator.
+ * @return std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>> A tuple of
+ * rmm::device_uvector objects for edge source vertex IDs and edge destination vertex IDs.
+ */
+template <typename vertex_t>
+std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<vertex_t>>
+generate_erdos_renyi_graph_edgelist(raft::handle_t const& handle,
+                                    size_t num_vertices,
+                                    float p,
+                                    bool symmetrize = false,
+                                    uint64_t seed = 0);
+
+/**
+ * @brief translate vertex ids in a graph by a specified offset
+ *
+ * Given an edgelist for a graph, translate all vertex ids by the given offset.
+ * This translation is done in place.
+ *
+ * If executed in a multi-gpu context (handle comms has been initialized)
+ * each GPU will operate only on its subset of data.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param d_src_v Vector of source vertices
+ * @param d_dst_v Vector of destination vertices
+ * @param vertex_id_offset Offset to add to each vertex id
+ */
+template <typename vertex_t>
+void translate_vertex_ids(raft::handle_t const &handle,
```

Comment:
This function is unnecessary if we take (vertex_first, vertex_last) instead of num_vertices